plugins/elasticsearch/elasticsearch.go

```diff
@@ -166,43 +212,66 @@ func (o *ElasticSearchOutput) receiver(or OutputRunner, wg *sync.WaitGroup) {
 				outBatch = append(outBatch, outBytes...)
 				if count = count + 1; o.bulkIndexer.CheckFlush(count, len(outBatch)) {
```

Comment:
Yup, as you discovered, the other goroutines might cache the value they have and not get updates when the mutation happens elsewhere. For higher level data structures there's RWMutex to distinguish btn read locks and write locks, but in this case the atomic API should suffice.